core/src/main/java/org/apache/servicecomb/core/Invocation.java

```diff
@@ -482,4 +489,50 @@ public boolean isThirdPartyInvocation() {
 
     return future;
   }
+
+  public void ensureInvocationNotTimeout() throws InvocationException {
```

Comment:
create a new handler to implement this feature?

Reply:
Now using event to check timeout